Realm/Tests/RealmConfigurationTests.mm

```diff
@@ -118,26 +117,27 @@ - (void)testDefaultConfiugrationUsesValueSemantics {
 
 - (void)testDefaultRealmUsesDefaultConfiguration {
     RLMRealmConfiguration *config = [RLMRealmConfiguration defaultConfiguration];
-    @autoreleasepool { XCTAssertEqualObjects(RLMRealm.defaultRealm.path, config.path); }
+    @autoreleasepool { XCTAssertEqualObjects(RLMRealm.defaultRealm.configuration.path, config.path); }
 
     config.path = RLMTestRealmPath();
-    @autoreleasepool { XCTAssertNotEqualObjects(RLMRealm.defaultRealm.path, config.path); }
+    @autoreleasepool { XCTAssertNotEqualObjects(RLMRealm.defaultRealm.configuration.path, config.path); }
     RLMRealmConfiguration.defaultConfiguration = config;
-    @autoreleasepool { XCTAssertEqualObjects(RLMRealm.defaultRealm.path, config.path); }
+    @autoreleasepool { XCTAssertEqualObjects(RLMRealm.defaultRealm.configuration.path, config.path); }
 
     config.inMemoryIdentifier = @"default";
     RLMRealmConfiguration.defaultConfiguration = config;
+    __block NSString *realmPath = nil;
     @autoreleasepool {
-        RLMRealm *realm = RLMRealm.defaultRealm;
-        XCTAssertTrue([realm.path hasSuffix:@"/default"]);
-        XCTAssertTrue([realm.path hasPrefix:NSTemporaryDirectory()]);
+        realmPath = @(config.config.path.c_str());
+        XCTAssertTrue([realmPath hasSuffix:@"/default"]);
+        XCTAssertTrue([realmPath hasPrefix:NSTemporaryDirectory()]);
     }
 
     config.schemaVersion = 1;
     RLMRealmConfiguration.defaultConfiguration = config;
     @autoreleasepool {
-        RLMRealm *realm = RLMRealm.defaultRealm;
-        XCTAssertEqual(1U, [RLMRealm schemaVersionAtPath:realm.path error:nil]);
+        __unused RLMRealm *realm = RLMRealm.defaultRealm;
```

Comment:
actually it was necessary to access the Realm here to cause the migration to happen, but I've refactored this to no longer have an unused variable.